### ISSUE TITLE
On Mac M2 "cannot import name '_BackgroundSyncer' from 'ray.tune.syncer'" error when using hyperopt 

### ISSUE THREAD

**Issue by sbocconi** [ludwig-ai/ludwig#3669]
**Describe the bug**
Running 
```
ludwig hyperopt --config rotten_tomatoes.yaml --dataset rotten_tomatoes.csv
```
on Mac gives
```
cannot import name '_BackgroundSyncer' from 'ray.tune.syncer'
```

**To Reproduce**

1. Install ludwig on Mac
1. Follow the tutorial and get to the step described in the [hyperoptimization](https://ludwig.ai/latest/getting_started/hyperopt/)
1. run ```ludwig hyperopt --config rotten_tomatoes.yaml --dataset rotten_tomatoes.csv```
1. You get ```cannot import name '_BackgroundSyncer' from 'ray.tune.syncer'```

A temporary fix (although the "internal" would hint at something that it should not be done") is the following:
in file ```ludwig/hyperopt/syncer.py ``` substitute 
```
from ray.tune.syncer import _BackgroundSyncer
```
with
```
from ray.train._internal.syncer import _BackgroundSyncer
```

**Expected behavior**
The command should run without import errors

**Environment (please complete the following information):**

- OS: macOS 
- Version: Ventura 13.5.2 (22G91)
- Python version: 3.11.0
- Ludwig version:  v0.8.4 - ludwig cli

**Comment by tgaddair** [comment]
Hey @sbocconi, thanks for raising this issue. Can you also share the version of Ray you're currently using? I wonder if this is an issue specific to a newer version of Ray.

**Comment by sbocconi** [comment]
From pip freeze:
```
ray==2.7.0
```

**Comment by arnavgarg1** [comment]
Hey @sbocconi! Ludwig is currently only compatible with Ray 2.3.1 and below. Are you able to downgrade Ray to 2.3.1 and give it a try?

**Comment by sbocconi** [comment]
Hey @arnavgarg1, thanks for your reply. I had to downgrade also Python to 3.10 as Ray 2.3.1 was not available for 3.11, but it seems to run now, thank you for your support.

Unfortunately the use of Ludwig and other DL application on Mac is greatly hindered by the fact that ```bitsandbytes``` is not available for Mac silicon, but this is outside your responsibility.

**Comment by arnavgarg1** [comment]
@sbocconi That sounds right! We only support Py38, 39 and 310, as well as Ray <= 2.3.1. It is true that bitsandbytes isn't available for Mac silicon - could be worth flagging with the bitsandbytes team to see if they can build appropriate wheels
